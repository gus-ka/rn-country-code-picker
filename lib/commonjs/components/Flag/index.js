"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const Flag = _ref => {
  let {
    flag
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: _styles.default.container
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Image, {
    style: _styles.default.image,
    source: flag,
    resizeMethod: "scale",
    testID: "flag-image"
  }));
};
var _default = Flag;
exports.default = _default;
//# sourceMappingURL=index.js.map