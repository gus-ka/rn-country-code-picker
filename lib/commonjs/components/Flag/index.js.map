{"version":3,"names":["_react","_interopRequireDefault","require","_reactNative","_styles","obj","__esModule","default","Flag","_ref","flag","createElement","View","style","styles","container","Image","image","source","resizeMethod","testID","_default","exports"],"sources":["index.tsx"],"sourcesContent":["import React from 'react';\nimport { Image, View } from 'react-native';\nimport { IFlagProps } from '../../types';\nimport styles from './styles';\n\nconst Flag: React.FC<IFlagProps> = ({ flag }) => {\n  return (\n    <View style={styles.container}>\n      <Image style={styles.image} source={flag} resizeMethod=\"scale\" testID=\"flag-image\" />\n    </View>\n  );\n};\n\nexport default Flag;\n"],"mappings":";;;;;;AAAA,IAAAA,MAAA,GAAAC,sBAAA,CAAAC,OAAA;AACA,IAAAC,YAAA,GAAAD,OAAA;AAEA,IAAAE,OAAA,GAAAH,sBAAA,CAAAC,OAAA;AAA8B,SAAAD,uBAAAI,GAAA,WAAAA,GAAA,IAAAA,GAAA,CAAAC,UAAA,GAAAD,GAAA,KAAAE,OAAA,EAAAF,GAAA;AAE9B,MAAMG,IAA0B,GAAGC,IAAA,IAAc;EAAA,IAAb;IAAEC;EAAK,CAAC,GAAAD,IAAA;EAC1C,oBACET,MAAA,CAAAO,OAAA,CAAAI,aAAA,CAACR,YAAA,CAAAS,IAAI;IAACC,KAAK,EAAEC,eAAM,CAACC;EAAU,gBAC5Bf,MAAA,CAAAO,OAAA,CAAAI,aAAA,CAACR,YAAA,CAAAa,KAAK;IAACH,KAAK,EAAEC,eAAM,CAACG,KAAM;IAACC,MAAM,EAAER,IAAK;IAACS,YAAY,EAAC,OAAO;IAACC,MAAM,EAAC;EAAY,CAAE,CAChF,CAAC;AAEX,CAAC;AAAC,IAAAC,QAAA,GAEab,IAAI;AAAAc,OAAA,CAAAf,OAAA,GAAAc,QAAA"}