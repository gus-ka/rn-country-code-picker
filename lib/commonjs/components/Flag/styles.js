"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _reactNative = require("react-native");
var _theme = require("../../theme");
var _default = _reactNative.StyleSheet.create({
  container: {
    width: 40,
    height: 30,
    borderRadius: _theme.borderRadius.s,
    marginRight: _theme.spacing.s
  },
  image: {
    width: '100%',
    height: '100%'
  }
});
exports.default = _default;
//# sourceMappingURL=styles.js.map