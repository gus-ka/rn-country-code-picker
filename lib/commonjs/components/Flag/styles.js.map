{"version":3,"names":["_reactNative","require","_theme","_default","StyleSheet","create","container","width","height","borderRadius","s","marginRight","spacing","image","exports","default"],"sources":["styles.ts"],"sourcesContent":["import { StyleSheet } from 'react-native';\nimport { borderRadius, spacing } from '../../theme';\n\nexport default StyleSheet.create({\n  container: {\n    width: 40,\n    height: 30,\n    borderRadius: borderRadius.s,\n    marginRight: spacing.s,\n  },\n  image: {\n    width: '100%',\n    height: '100%',\n  },\n});\n"],"mappings":";;;;;;AAAA,IAAAA,YAAA,GAAAC,OAAA;AACA,IAAAC,MAAA,GAAAD,OAAA;AAAoD,IAAAE,QAAA,GAErCC,uBAAU,CAACC,MAAM,CAAC;EAC/BC,SAAS,EAAE;IACTC,KAAK,EAAE,EAAE;IACTC,MAAM,EAAE,EAAE;IACVC,YAAY,EAAEA,mBAAY,CAACC,CAAC;IAC5BC,WAAW,EAAEC,cAAO,CAACF;EACvB,CAAC;EACDG,KAAK,EAAE;IACLN,KAAK,EAAE,MAAM;IACbC,MAAM,EAAE;EACV;AACF,CAAC,CAAC;AAAAM,OAAA,CAAAC,OAAA,GAAAZ,QAAA"}