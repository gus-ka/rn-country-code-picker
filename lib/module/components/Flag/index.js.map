{"version":3,"names":["React","Image","View","styles","Flag","_ref","flag","createElement","style","container","image","source","resizeMethod","testID"],"sources":["index.tsx"],"sourcesContent":["import React from 'react';\nimport { Image, View } from 'react-native';\nimport { IFlagProps } from '../../types';\nimport styles from './styles';\n\nconst Flag: React.FC<IFlagProps> = ({ flag }) => {\n  return (\n    <View style={styles.container}>\n      <Image style={styles.image} source={flag} resizeMethod=\"scale\" testID=\"flag-image\" />\n    </View>\n  );\n};\n\nexport default Flag;\n"],"mappings":"AAAA,OAAOA,KAAK,MAAM,OAAO;AACzB,SAASC,KAAK,EAAEC,IAAI,QAAQ,cAAc;AAE1C,OAAOC,MAAM,MAAM,UAAU;AAE7B,MAAMC,IAA0B,GAAGC,IAAA,IAAc;EAAA,IAAb;IAAEC;EAAK,CAAC,GAAAD,IAAA;EAC1C,oBACEL,KAAA,CAAAO,aAAA,CAACL,IAAI;IAACM,KAAK,EAAEL,MAAM,CAACM;EAAU,gBAC5BT,KAAA,CAAAO,aAAA,CAACN,KAAK;IAACO,KAAK,EAAEL,MAAM,CAACO,KAAM;IAACC,MAAM,EAAEL,IAAK;IAACM,YAAY,EAAC,OAAO;IAACC,MAAM,EAAC;EAAY,CAAE,CAChF,CAAC;AAEX,CAAC;AAED,eAAeT,IAAI"}