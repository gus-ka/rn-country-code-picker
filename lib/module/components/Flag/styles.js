import { StyleSheet } from 'react-native';
import { borderRadius, spacing } from '../../theme';
export default StyleSheet.create({
  container: {
    width: 40,
    height: 30,
    borderRadius: borderRadius.s,
    marginRight: spacing.s
  },
  image: {
    width: '100%',
    height: '100%'
  }
});
//# sourceMappingURL=styles.js.map