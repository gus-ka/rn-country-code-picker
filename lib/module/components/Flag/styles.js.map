{"version":3,"names":["StyleSheet","borderRadius","spacing","create","container","width","height","s","marginRight","image"],"sources":["styles.ts"],"sourcesContent":["import { StyleSheet } from 'react-native';\nimport { borderRadius, spacing } from '../../theme';\n\nexport default StyleSheet.create({\n  container: {\n    width: 40,\n    height: 30,\n    borderRadius: borderRadius.s,\n    marginRight: spacing.s,\n  },\n  image: {\n    width: '100%',\n    height: '100%',\n  },\n});\n"],"mappings":"AAAA,SAASA,UAAU,QAAQ,cAAc;AACzC,SAASC,YAAY,EAAEC,OAAO,QAAQ,aAAa;AAEnD,eAAeF,UAAU,CAACG,MAAM,CAAC;EAC/BC,SAAS,EAAE;IACTC,KAAK,EAAE,EAAE;IACTC,MAAM,EAAE,EAAE;IACVL,YAAY,EAAEA,YAAY,CAACM,CAAC;IAC5BC,WAAW,EAAEN,OAAO,CAACK;EACvB,CAAC;EACDE,KAAK,EAAE;IACLJ,KAAK,EAAE,MAAM;IACbC,MAAM,EAAE;EACV;AACF,CAAC,CAAC"}