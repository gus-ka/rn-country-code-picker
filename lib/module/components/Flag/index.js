import React from 'react';
import { Image, View } from 'react-native';
import styles from './styles';
const Flag = _ref => {
  let {
    flag
  } = _ref;
  return /*#__PURE__*/React.createElement(View, {
    style: styles.container
  }, /*#__PURE__*/React.createElement(Image, {
    style: styles.image,
    source: flag,
    resizeMethod: "scale",
    testID: "flag-image"
  }));
};
export default Flag;
//# sourceMappingURL=index.js.map