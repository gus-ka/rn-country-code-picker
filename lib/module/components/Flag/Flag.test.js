import React from 'react';
import { render } from '@testing-library/react-native';
import mockCountry from '../../__mocks__/country';
import Flag from './';
test('renders correctly', () => {
  const mockFlag = mockCountry.flag;
  const {
    toJSON,
    getByTestId
  } = render( /*#__PURE__*/React.createElement(Flag, {
    flag: mockFlag
  }));
  const rendered = toJSON();
  const image = getByTestId('flag-image');
  expect(image.props.source).toBe(mockFlag);
  expect(rendered).toMatchSnapshot();
});
//# sourceMappingURL=Flag.test.js.map