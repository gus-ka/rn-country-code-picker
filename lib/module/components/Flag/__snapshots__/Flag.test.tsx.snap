// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 4,
      "height": 30,
      "marginRight": 8,
      "width": 40,
    }
  }
>
  <Image
    resizeMethod="scale"
    source={
      Object {
        "testUri": "../../../src/assets/flags/tr.png",
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    testID="flag-image"
  />
</View>
`;
